src/components/[guild]/JoinModal/components/CompleteCaptchaJoinStep.tsx

```diff
@@ -15,8 +15,10 @@ const CompleteCaptchaJoinStep = (): JSX.Element => {
     ?.filter((req) => req.type === "CAPTCHA")
     .map((req) => req.id)
 
-  const { data: accesses } = useAccess()
-  const requirementAccesses = accesses?.flatMap((access) => access.requirements)
+  const { membership } = useMembership()
+  const requirementAccesses = membership?.roles?.flatMap((role) =>
```

Comment:
is it intentional that we filter only true accesses here (and in every place where we check requirementAccesses)? Maybe it isn't necessary, since we check if `req.access` is true in `isDone` logic

Reply:
Nice catch! It might not be needed then! Looking at the diff, it indeed seems like, that `requirementAccesses` included all the requirements, not just the satisfied ones
Changed in https://github.com/guildxyz/guild.xyz/pull/1048/commits/ddcf71b4d77662b47692b6b9ebc341f92c6896e9!